### PR TITLE
[REVIEW] Fix `columns` ordering issue in parquet reader

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #10062 

This PR fixes issue where the order of `columns` and parquet metadata columns(i.e., `meta['columns']`) can differ and both are not guaranteed to be in the same order always. In this PR, removed the code that has this assumption and created a new dict that contains the metadata of columns which are later used to update the column metadata in dataframe.

### CODE REVIEW

File: python/cudf/cudf/_lib/parquet.pyx [hunk 1]
```diff
@@ -200,12 +200,19 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
 
     update_struct_field_names(df, c_out_table.metadata.schema_info)
 
-    # update the decimal precision of each column
     if meta is not None:
-        for col, col_meta in zip(column_names, meta["columns"]):
+        # Book keep each column metadata as the order
+        # of `meta["columns"]` and `column_names` are not
+        # guaranteed to be deterministic and same always.
+        meta_data_per_column = {}
```
Comment (rgsl888prabhu) [inline python/cudf/cudf/_lib/parquet.pyx:207]: Can we do something like this 

meta_data_per_column = {col_meta['name']:col_meta for col_meta in meta["columns"]}

**Review by rgsl888prabhu — APPROVED** [review]
A small suggestion, rest looks good

File: python/cudf/cudf/_lib/parquet.pyx (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/_lib/parquet.pyx]: Yup, done 👍

**Comment by vuule** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
`dask_cudf.tests.test_groupby.test_groupby_basic[True-var]` failed about five times in a row. I know it can pass since it passed once on my PR, and on #10067. Might be better to admin merge these instead to retrying so many times.

**Comment by galipremsagar** [comment]
> `dask_cudf.tests.test_groupby.test_groupby_basic[True-var]` failed about five times in a row. I know it can pass since it passed once on my PR, and on #10067. Might be better to admin merge these instead to retrying so many times.

This was a cupy falky issue, https://github.com/rapidsai/cudf/pull/10071 has a fix for it. I'll get it merged ASAP.

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**